common/src/sql/dbinit.sql

```diff
@@ -244,6 +244,7 @@ CREATE TABLE omicron.public.silo (
 
     discoverable BOOL NOT NULL,
     user_provision_type omicron.public.user_provision_type NOT NULL,
+    admin_group_name TEXT,
```

Comment:
If it's not mutable, then the admin group name has to be constant for the lifetime of the Silo, meaning the Silo would have to be re-created for the admin group name to change. However, if this group is *only* used to log in and bootstrap the actual permission structure that you want for the silo, it can be more transient, and it could even be deleted by the user when they're done. But that's at odds with the idea that all users and groups will come from their IDP.

I'm ok leaving this mutable to support changing the IDP's group name, having the user re-login to JIT a new admin group, and continuing on.

Reply:
I'm confused about this.  I would have expected that we don't store the admin group name on the Silo at all, but rather we just use it when creating the Silo to create the initial Silo Group and grant it the "Silo Admin" role.  If someone wants to change which IdP group should get that role, they can just grant the "Silo Admin" privilege to a different IdP group instead.  How is that at odds with the idea that users and groups come from the IdP?

Reply:
Sorry, what I meant was that if users are deleting groups manually, then *that* is at odds with the idea that all users and groups will come from their IDP.

Note this is a special case of silo group creation: it occurs during the initial silo setup, not as a result of JIT from login details, and it assigns permissions, which normal group creation doesn't do. Without any way of triggering this special case again, there could be drift between the IdP group name and the silo group name.

I want to avoid the scenario where the group name changes in the IdP, the admin user re-logs in (JITing this new group with no associated permissions), and not being able to fix it due to lack of permission. In order for this scenario to *not* occur though, it requires some action before the new permissionless group is created. Users must create this group and associated permissions themselves, before the rename occurs and when they still have their permissions (from the group). If they do this, the group is created by them, and not by the login itself. Unfortunately, this is also at odds with the idea that all users and groups come from the IdP, and it's a direct result of the fact that no permissions come from the IdP.

Luckily the fix for this is simple - rename the admin group back to what it was in the IdP, then re-login. Because the fix is simple, I'm inclined to what you're suggesting, which is to not store admin_group_name - which is done in 54e246e4